compiler/rustc_lint/src/unused.rs

```diff
@@ -517,16 +517,14 @@ trait UnusedDelimLint {
         right_pos: Option<BytePos>,
     ) {
         let spans = match value.kind {
-            ast::ExprKind::Block(ref block, None) if block.stmts.len() > 0 => {
-                let start = block.stmts[0].span;
-                let end = block.stmts[block.stmts.len() - 1].span;
-                if let Some(start) = start.find_ancestor_inside(value.span)
-                    && let Some(end) = end.find_ancestor_inside(value.span)
+            ast::ExprKind::Block(ref block, None) if block.stmts.len() == 1 => {
```

Comment:
According to the following comment, the block always has one statement.

https://github.com/rust-lang/rust/blob/101e1822c3e54e63996c8aaa014d55716f3937eb/compiler/rustc_lint/src/unused.rs#L992-L998